Assets/MRTK/SDK/Features/UX/Scripts/Slate/HandInteractionPanZoom.cs

```diff
@@ -415,6 +415,55 @@ private void UpdateUVMapping()
                         // This is where zoom is applied if Active
                         uvs[i] = ((uvs[i] - scaleUVCentroid) / scaleUVDelta) + scaleUVCentroid;
                     }
+
+                    // Put pan limits
+                    if (!unlimitedPan)
+                    {
+                        var correctX = false;
+                        var correctY = false;
+                        var positionCorrection = Vector2.zero;
+
+                        for (int i = 0; i < uvs.Count; ++i)
+                        {
+                            var uv = uvs[i];
+
+                            if (!correctX)
+                            {
+                                if (uv.x > tiling.x * maxPanHorizontal)
+                                {
+                                    positionCorrection.x = tiling.x * maxPanHorizontal - uv.x;
+                                    correctX = true;
+                                }
+                                else if (uv.x < -(tiling.x * maxPanHorizontal))
+                                {
+                                    positionCorrection.x = tiling.x * (-maxPanHorizontal) - uv.x;
+                                    correctX = true;
+                                }
+                            }
+
+                            if (!correctY)
+                            {
+                                if (uv.y > tiling.y * maxPanVertical)
+                                {
+                                    positionCorrection.y = tiling.y * maxPanVertical - uv.y;
+                                    correctY = true;
+                                }
+                                else if (uv.y < -(tiling.y * maxPanVertical))
+                                {
+                                    positionCorrection.y = tiling.y * (-maxPanVertical) - uv.y;
+                                    correctY = true;
+                                }
+                            }
+                        }
+
+                        if (correctX || correctY)
+                        {
+                            for (int i = 0; i < uvs.Count; ++i)
+                            {
+                                uvs[i] += positionCorrection;
+                            }
+                        }
+                    }
```

Comment:
I have noticed there is already a check for pan limits before applying uv deltas. Could you please try avoiding the duplicates and combine the checking code and its variables.

`// Test for pan limits
            Vector2 uvDelta = new Vector2(totalUVOffset.x, -totalUVOffset.y);
            if (!unlimitedPan)
            {
                bool xLimited = false;
                bool yLimited = false;
                for (int i = 0; i < uvs.Count; ++i)
                {
                    uvTestValue = uvs[i] - uvDelta;
                    if (uvTestValue.x > tiling.x * maxPanHorizontal || uvTestValue.x < -(tiling.x * maxPanHorizontal))
                    {
                        xLimited = true;
                    }
                    if (uvTestValue.y > tiling.y * maxPanVertical || uvTestValue.y < -(tiling.y * maxPanVertical))
                    {
                        yLimited = true;
                    }
                }`